libpod/boltdb_state.go

```diff
@@ -458,14 +455,28 @@ func (s *BoltState) LookupContainer(idOrName string) (*Container, error) {
 			}
 			return errors.Wrapf(define.ErrNoSuchCtr, "no container with name or ID %s found", idOrName)
 		}
-
-		return s.getContainerFromDB(id, ctr, ctrBucket)
+		return nil
 	})
+
+	if err != nil {
+		return []byte{}, err
+	}
+
+	retID := make([]byte, len(id))
+	copy(retID, id)
+
+	return retID, nil
+}
+
+// LookupContainer retrieves a container from the state by full or unique
+// partial ID or name
+func (s *BoltState) LookupContainer(idOrName string) (*Container, error) {
+	id, err := s.LookupContainerID(idOrName)
 	if err != nil {
 		return nil, err
 	}
 
-	return ctr, nil
+	return s.Container(string(id))
```

Comment:
We should try and avoid using two database transactions here - LookupContainer is used a lot. Sharing the code is still a good goal, but we should try and use helpers within the transaction itself.